src/main/java/uk/co/bconline/ndelius/service/impl/UserEntityServiceImpl.java

```diff
@@ -85,7 +87,14 @@ public boolean usernameExists(String username)
 	@Override
 	public Optional<StaffEntity> getStaffByStaffCode(String code)
 	{
-		return staffRepository.findByCode(code);
+		try
+		{
+			return staffRepository.findByCodeAndEndDateIsNull(code).or(() -> staffRepository.findByCode(code));
+		}
+		catch (IncorrectResultSizeDataAccessException e)
+		{
+			throw new AppException("Multiple active or multiple inactive staff records found for this officer code - " + code);
```

Comment:
```suggestion
			throw new AppException("Unable to select a unique Staff Record for code: " + code);
```

Slightly more concise error message, what do you think?